runtime/src/main/kotlin/kotlin/native/concurrent/SharedRef.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+
+@SymbolName("Kotlin_SharedRef_createSharedRef")
+external private fun createSharedRef(value: Any): NativePtr
+
+/**
+ * A frozen shared reference to a Kotlin object
+ *
+ * Can be safely passed between workers, but can only be accessed on the worker it was created on,
+ * unless the referred object is frozen too
+ *
+ * Note: Garbage collector currently cannot free any reference cycles with [SharedRef] or [DisposableSharedRef] in them.
+ * Consider using [DisposableSharedRef] to manually resolve cycles with explicit calls to [DisposableSharedRef.dispose]
+ */
+@Frozen
+@NoReorderFields
+@ExportTypeInfo("theSharedRefTypeInfo")
+public class SharedRef<out T : Any>(value: T) {
```

Comment:
`...Ref` is inconsistent with other `...Reference` types, like `AtomicReference`, `WeakReference`